test/generalizedgaussian.jl

```diff
@@ -0,0 +1,234 @@
+    -0.9899 0.1611117466391557;
+    -0.9495 0.1711844697448073;
+    -0.9091 0.1816510704434489;
+    -0.8687 0.1925092128511123;
+    -0.8283 0.2037551726565467;
+    -0.7879 0.2153838049289052;
+    -0.7475 0.2273885202046074;
+    -0.7071 0.2397612693401354;
+    -0.6667 0.2524925375469229;
+    -0.6263 0.2655713479467356;
+    -0.5859 0.2789852749018126;
+    -0.5455 0.2927204672844635;
+    -0.5051 0.3067616817568189;
+    -0.4646 0.3210923260341274;
+    -0.4242 0.3356945120055987;
+    -0.3838 0.3505491184865381;
+    -0.3434 0.3656358632757433;
+    -0.3030 0.3809333840941239;
+    -0.2626 0.3964193278856127;
+    -0.2222 0.4120704478709425;
+    -0.1818 0.4278627076600358;
+    -0.1414 0.4437713916507750;
+    -0.1010 0.4597712208719139;
+    -0.0606 0.4758364733668302;
+    -0.0202 0.4919411081636175;
+    0.0202 0.5080588918363826;
+    0.0606 0.5241635266331698;
+    0.1010 0.5402287791280863;
+    0.1414 0.5562286083492251;
+    0.1818 0.5721372923399641;
+    0.2222 0.5879295521290574;
+    0.2626 0.6035806721143873;
+    0.3030 0.6190666159058762;
+    0.3434 0.6343641367242567;
+    0.3838 0.6494508815134618;
+    0.4242 0.6643054879944013;
+    0.4646 0.6789076739658725;
+    0.5051 0.6932383182431812;
+    0.5455 0.7072795327155365;
+    0.5859 0.7210147250981875;
+    0.6263 0.7344286520532644;
+    0.6667 0.7475074624530771;
+    0.7071 0.7602387306598648;
+    0.7475 0.7726114797953927;
+    0.7879 0.7846161950710949;
+    0.8283 0.7962448273434533;
+    0.8687 0.8074907871488877;
+    0.9091 0.8183489295565511;
+    0.9495 0.8288155302551927;
+    0.9899 0.8388882533608444;
+    1.0303 0.8485661114948515;
+    1.0707 0.8578494187361720;
+    1.1111 0.8667397370974946;
+    1.1515 0.8752398172113575;
+    1.1919 0.8833535339399387;
+    1.2323 0.8910858176405667;
+    1.2727 0.8984425818284343;
+    1.3131 0.9054306479787271;
+    1.3535 0.9120576682028299;
+    1.3939 0.9183320465178874;
+    1.4343 0.9242628594063882;
+    1.4747 0.9298597763332334;
+    1.5152 0.9351329808526656;
+    1.5556 0.9400930928972280;
+    1.5960 0.9447510927963858;
+    1.6364 0.9491182475243711;
+    1.6768 0.9532060396260427;
+    1.7172 0.9570260992168560;
+    1.7576 0.9605901393992229;
+    1.7980 0.9639098953833416;
+    1.8384 0.9669970675467084;
+    1.8788 0.9698632686136334;
+    1.9192 0.9725199750847831;
+    1.9596 0.9749784829975958;
+    2.0000 0.9772498680518207;
+]
+
+# CDF test.
+for i=1:size(cdftest, 1)
+    @test isapprox(cdf(d, cdftest[i, 1]), cdftest[i, 2] ; atol = 1e-6)
+end
+
+# PDF test.
+for i=1:size(pdftest, 1)
+    @test isapprox(pdf(d, pdftest[i, 1]), pdftest[i, 2] ; atol = 1e-6)
+end
+
+@test mean(d) == 0
+@test median(d) == 0
+@test mode(d) == 0
+@test var(d) == 1 # unity variance with shape 2 and scale √2
+@test std(d) == 1
+@test skewness(d) == 0
+@test kurtosis(d) ≈ 0
+@test entropy(d) ≈ 1.418938533204673  #[1]
+# [1] Nadarajah, Saralees (September 2005).  "A generalized normal distribution".
```

Comment:
is the reference used for the implementation or the tested values only?

Reply:
The reference was for implementation, I’ll admit it is a bit haphazardly placed. I can move the reference to the src file where it is calculated instead?

Reply:
Yes, either in the docstring of the Distribution or of the relevant function if it's just for one of the functions